finality-aleph/src/new_network/connection_manager/session.rs

```diff
@@ -186,7 +216,7 @@ impl Handler {
 mod tests {
```

Comment:
I might have just missed it or author's intention was different (was it?) - can you upgrade a non-validator's handler to a validator's handler. Anywa, there are no test cases checking such transition (or I just missed them).

Reply:
Added case to forbid such change and tests that it is correct